src/mpc-hc/PlayerSeekBar.cpp

```diff
@@ -853,3 +919,59 @@ void CPlayerSeekBar::OnCaptureChanged(CWnd* pWnd)
         m_pMainFrame->UpdateControlState(CMainFrame::UPDATE_CHILDVIEW_CURSOR_HACK);
     }
 }
+
+REFERENCE_TIME CPlayerSeekBar::CalculatePosition(const CPoint point) {
+    REFERENCE_TIME pos = -1;
+    const CRect r = GetChannelRect();
+
+    if (point.x < r.left) {
+        pos = 0;
+    } else if (point.x >= r.right) {
+        pos = m_rtStop;
+    } else if (m_rtStop > 0) {
+        const LONG w = r.right - r.left;
+        pos = (m_rtStop * (point.x - r.left) + (w / 2)) / w;
+    }
+
+    return pos;
+}
+
+void CPlayerSeekBar::MoveThumbPreview(const CPoint point) {
+    REFERENCE_TIME pos = CalculatePosition(point);
+
+    if (pos >= 0) {
+        if (AfxGetAppSettings().bFastSeek ^ (GetKeyState(VK_SHIFT) < 0)) {

```

Comment:
Should we maybe always use fast keyframe seek for the preview? But still allow to override it by pressing shift?

Reply:
So would that just be !bFastSeek ^ ... ?

Reply:
No, I mean completely removing bFastSeek from the IF. So always do fast seek in the preview, regardless of the setting for playback seeking.

Reply:
OK, so the shift only applies to the real seeking, and not the preview?

Reply:
The way I read that code, it's saying to do fastseek when:
fastseek is enabled and you aren't holding shift
fastseek is disabled and you are holding shift

So if we remove the boolean, that means it will only fastseek if we hold shift (unless we invert the getkeystate).